### PR TITLE
Tbd user assignment

### LINKED ISSUE #75: Need "TBD" as a person
Sometimes we dont know who exactly will be working a project... so we need to stub a "person" into the plan.

Can we please add a "TBD" option in the Add Person action?
And then when we know who it will be, we click on the TBD label in the plan, and can assign it to a real person.

### PR DESCRIPTION
**Pull request by HankC138**
An assignment can now be created for an undetermined user and assigned later. Closes #75

### CODE REVIEW

File: app/components/projectAssignment/workWeekInput.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ interface WorkWeekInputProps {
 	assignment?: AssignmentType;
 	cweek: number | null;
 	year: number | null;
+	isUserTBD:boolean;
```
Comment (fermion) [inline app/components/projectAssignment/workWeekInput.tsx:14]: nit: space between colon and type? We should probably set up some linting rules for things like this.

File: app/components/userAssignment/tempProjectLabel.tsx [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+import Image from "next/image";
+import React from "react";
+
+interface TempLabelProps {
+	assignment: AssignmentType;
+	tempProjectOpen: boolean;
+	setTempProjectOpen: React.Dispatch<React.SetStateAction<boolean>>;
+}
+export const TempProjectLabel = ({ assignment, tempProjectOpen, setTempProjectOpen }: TempLabelProps) => {
+	// const newProjectInitialValues = {
```
Comment (fermion) [inline app/components/userAssignment/tempProjectLabel.tsx:11]: Delete?

File: app/people/[userId]/page.tsx [hunk 3]
```diff
@@ -46,35 +47,92 @@ const UserPage: React.FC = () => {
...
-	}, [clientSide, userList, params.name, viewsFilter]);
+	}, [clientSide, userList, params.userId, viewsFilter]);
 
 	if (!userList) return <LoadingSpinner />;
 	const onClose = () => setAddAssignmentVisible(false);
 	const onComplete = () => {
 		setAddAssignmentVisible(false);
 	};
+
+	// const addNewAssignment = (newAssignment: AssignmentType) => {
```
Comment (fermion) [inline app/people/[userId]/page.tsx:65]: Delete?

**Review by fermion — COMMENTED** [review]
Same feedback here. Some dead code to delete, a few nits but looks fine to me otherwise. Would defer to @MyStarrySpace @zhukovdigital for anything beyond a cursory look.

File: app/people/[userId]/page.tsx (hunk 3 above)
Comment (HankC138) [reply app/people/[userId]/page.tsx:65]: deleted

File: app/components/addAssignmentModal.tsx [hunk 4]
```diff
@@ -110,17 +110,17 @@ const AddAssignment = () => {
...
-		}
-		if (values.userId) {
-			const foundUser = userList?.find(
-				({ id }: UserType) => id === values.userId
-			);
-			if (!foundUser) {
-				errors.userId = "Must select a valid User";
-			}
-		}
+		// if (!values.userId) {
```
[the remark this answers is not in the document]
Comment (HankC138) [reply app/components/addAssignmentModal.tsx:113]: removed user validation on form

File: app/components/projectAssignment/workWeekInput.tsx (hunk 1 above)
Comment (HankC138) [reply app/components/projectAssignment/workWeekInput.tsx:14]: added space

File: app/components/userAssignment/tempProjectLabel.tsx (hunk 2 above)
Comment (HankC138) [reply app/components/userAssignment/tempProjectLabel.tsx:11]: removed commented out code

**Review by fermion — APPROVED** [review]

**Pull request merged**